contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -6611,3 +6659,1195 @@ get_domain_typmodin(Type typ)
+											NULL, 1, rewrite_key);
+											
+			rewrite_tuple = systable_getnext(rewrite_scan);
+			
+			if (HeapTupleIsValid(rewrite_tuple))
+			{
+				Form_pg_rewrite rule_form = (Form_pg_rewrite) GETSTRUCT(rewrite_tuple);
+				dependentViewOid = rule_form->ev_class;
+			}
+			
+			systable_endscan(rewrite_scan);
+			
+			if (OidIsValid(dependentViewOid))
+			{
+				update_view_binding_to_weak(dependentViewOid, bbf_view_def_rel);
+				/* Recursively update views that depend on this view */
+				update_dependent_views_to_weak(dependentViewOid, depRel, bbf_view_def_rel);
+			}
+		}
+	}
+	systable_endscan(scan);
+	table_close(pg_rewrite_desc, AccessShareLock);
+}
+
+/*
+ * Check if a schema-bound view references any non-schema-bound views
+ *
+ * This function is called during view creation to enforce the SQL Server rule
+ * that schema-bound views cannot reference non-schema-bound views. It finds
+ * all views referenced in the query (including those in subqueries and sublinks)
+ * and checks if any of them are weak (non-schema-bound) views.
+ *
+ * The function works by:
+ * 1. Finding all views referenced in the query
+ * 2. Checking each view to see if it's a weak view
+ * 3. Throwing an error if any weak views are found
+ *
+ * Parameters:
+ * - viewParse: The query tree for the view being created
+ *
+ * Returns:
+ * - true if no weak view dependencies are found
+ * - false if weak view dependencies are found (after throwing an error)
+ */
+static bool
+check_view_binding_dependencies(Query *viewParse)
+{
+	List *dependent_views = NIL;
+	ListCell *lc;
+	int16 logical_dbid;
+	Relation bbf_view_def_rel;
+	HeapTuple btup;
+	const char *logical_schema_name;
+	bool is_strong_view = true;
+
+	if (pltsql_parse_result && pltsql_parse_result->body)
+	{
+		PLtsql_stmt_execsql *stmt = (PLtsql_stmt_execsql *) lsecond(pltsql_parse_result->body);
+		if (stmt && !weak_view_binding)
+			is_strong_view = true;
+		else
+			is_strong_view = stmt->is_schemabinding;
+	}
+
+	if(sql_dialect != SQL_DIALECT_TSQL)
+		return true;
+
+	if (weak_view_binding && !is_strong_view)
+        return true;
+
+	/* Find all view dependencies in the query, including those in subqueries*/
+	find_all_view_dependencies(viewParse, &dependent_views);
+	
+	foreach(lc, dependent_views)
+	{
+		Oid viewOid = lfirst_oid(lc);		
+		Oid schema_id = get_rel_namespace(viewOid);
+		char *schema_name = get_namespace_name(schema_id);
+		const char *view_name = get_rel_name(viewOid);
+		
+		if (schema_name && is_shared_schema(schema_name))
+		{
+			pfree(schema_name);
+			continue;
+		}
+		logical_dbid = InvalidDbid;
+		logical_schema_name = NULL;
+		
+		if (schema_name)
+		{
+			logical_dbid = get_dbid_from_physical_schema_name(schema_name, true);
+			logical_schema_name = get_logical_schema_name(schema_name, true);
+		}
+		
+		if (!schema_name || !logical_schema_name || !DbidIsValid(logical_dbid))
+		{
+			if (schema_name)
+				pfree(schema_name);
```

Comment:
After pfree, should also set schema_name = NULL, otherwise the pointer can point to nowhere.